### PR TITLE
supported distro detection for docker-in-docker script

### PR DESCRIPTION
**Pull request by joshspicer**
closes https://github.com/microsoft/vscode-dev-containers/issues/1367

Provide clearer error messaging and the ability to dynamically update supported distro releases for both OSS Moby and docker installation.

For example, trying to use moby dind with hirsute will provide the following guidance:
```bash
2022-03-17 16:50:05.406Z: #7 [3/3] RUN bash /tmp/scripts/docker-in-docker-debian.sh "true" "${USERNAME}" "true"
2022-03-17 16:50:05.429Z: #7 sha256:df641496a1c8141a6497bf1279fdb806b1d428b8c0c88e0272b4b0792f370b2a
2022-03-17 16:50:06.577Z: #7 1.206 DOCKER_IN_DOCKER_MOBY_ARCHIVE_VERSION_CODENAMES=buster bullseye bionic focal
2022-03-17 16:50:06.605Z: #7 1.206 (!) Unsupported  distribution version 'hirsute'. To resolve, either: (1) set feature option '"moby": false' , or (2) choose a compatible OS distribution
2022-03-17 16:50:06.632Z: #7 1.206 (!) Support distributions include:  buster bullseye bionic focal
2022-03-17 16:50:06.654Z: #7 ERROR: executor failed running [/bin/sh -c bash /tmp/scripts/docker-in-docker-debian.sh "true" "${USERNAME}" "true"]: exit code: 1
```

This set of codenames can be toggled as it reads from the common `get_common_setting` utilty function, which sources values from `script-library/shared/settings.env`

@chuxel  - Did I get all the supported docker/moby codenames right?

relevant to: https://github.com/microsoft/vscode-dev-containers/issues/1019

### CODE REVIEW

File: script-library/docker-in-docker-debian.sh [hunk 1]
```diff
@@ -15,14 +15,27 @@ USE_MOBY=${3:-"true"}
 DOCKER_VERSION=${4:-"latest"} # The Docker/Moby Engine + CLI should match in version
 MICROSOFT_GPG_KEYS_URI="https://packages.microsoft.com/keys/microsoft.asc"
 DOCKER_DASH_COMPOSE_VERSION="1"
+DOCKER_IN_DOCKER_MOBY_ARCHIVE_VERSION_CODENAMES="buster bullseye bionic focal"
+DOCKER_IN_DOCKER_LICENSED_ARCHIVE_VERSION_CODENAMES="buster bullseye bionic focal hirsute"
```
Comment (Chuxel) [inline script-library/docker-in-docker-debian.sh:19]: Ubuntu 21.10 "impish" also appears to be supported.

```suggestion
DOCKER_IN_DOCKER_LICENSED_ARCHIVE_VERSION_CODENAMES="buster bullseye bionic focal hirsute Impish"
```

**Comment by Chuxel** [comment]
Also - would it make sense to do this in docker-from-docker too?

File: migrations/readme-metadata-to-json.py [hunk 2]
```diff
@@ -106,17 +106,17 @@ def template(defId, displayName, description, categories, defType, platforms, op
         options = 'lc rc'
 
     subst = '''
-		lc
```
Comment (joshspicer) [inline migrations/readme-metadata-to-json.py:109]: Not relevant to this PR, oops :D

Comment (joshspicer) [reply migrations/readme-metadata-to-json.py:109]: Still want it though.

File: script-library/docker-debian.sh [hunk 3]
```diff
@@ -17,6 +17,9 @@ USE_MOBY=${5:-"true"}
 DOCKER_VERSION=${6:-"latest"}
 DOCKER_DASH_COMPOSE_VERSION=${7:-"v1"} # v1 or v2
 MICROSOFT_GPG_KEYS_URI="https://packages.microsoft.com/keys/microsoft.asc"
+DOCKER_DASH_COMPOSE_VERSION="1"
```
Comment (jbreiding) [inline script-library/docker-debian.sh:20]: I believe this line was removed in the previous commit.

Or is the pinning back to v1 intentional?

Comment (joshspicer) [reply script-library/docker-debian.sh:20]: Ah! Thank you for catching this - removed.

**Pull request merged**